### DISCUSSION TITLE
Foreground job killed with app

### DISCUSSION THREAD

**Discussion by sandeepc24** [shinyorg/shiny#986] [Q&A] [answered]
I have a service with `RunOnForegorund` flag set to true and when I kill the app (by swiping) I was expecting the service to keep running - but it does not, is this expected behaviour?

**Comment by aritchie** [comment, marked as answer]
Yes

**Reply by aritchie** [reply]
Since someone else asked about this - a job is NOT a service.  It will run on a timeslice provider by the android workmanager which does not run as a foreground service.

The "RunOnForeground" in a job specifies that the jobs should run periodically on a timer while the app itself is in the foreground.  If the app is swiped away, it will kill the entire process.
